app/src/main/java/org/oppia/app/player/audio/AudioFragment.kt

```diff
@@ -27,22 +27,20 @@ class AudioFragment : InjectableFragment() {
   }
 
   fun languageSelectionClicked() {
-    languageInterface = object : LanguageInterface {
-      override fun onLanguageSelected(currentLanguageCode: String) {
-        audioFragmentPresenter.languageSelected(currentLanguageCode)
-      }
-    }
+    showLanguageDialogFragment()
+  }
 
-    val previousFragment = fragmentManager?.findFragmentByTag(TAG_DIALOG)
+  private fun showLanguageDialogFragment() {
+    val previousFragment = fragmentManager?.findFragmentByTag(TAG_LANGUAGE_DIALOG)
     if (previousFragment != null) {
       fragmentManager?.beginTransaction()?.remove(previousFragment)?.commitNow()
     }
     val dialogFragment = LanguageDialogFragment.newInstance(
-      languageInterface,
-      getDummyAudioLanguageList(),
-      "en"
+      getDummyAudioLanguageList() as ArrayList<String>,
+      selectedLanguageCode
     )
-    dialogFragment.showNow(fragmentManager, TAG_DIALOG)
+    dialogFragment.setTargetFragment(this, REQUEST_CODE_DIALOG)
```

Comment:
This is cool, I didn't know about this. However, are we actually using the activity result? It doesn't seem like we need to set this here, but maybe I'm misunderstanding the intent.

Edit: ah I see you're setting this so that we can cast to the interface. If that's the only purpose of setting the target fragment, could we maybe instead use [getParentFragment()](https://developer.android.com/reference/android/support/v4/app/Fragment.html#getParentFragment())? That seems a bit cleaner since it avoids the extra onActivityResult behavior of setTargetFragment that we don't really need here.

Reply:
Actually, `REQUEST_CODE_DIALOG` inside `setTargetFragment` is not needed for us but `setTargetFragment` requires this parameter.

Regarding, `getParentFragment()` gives the reference to `StateFragment` rather than `AudioFragment` because of which I am not able to use `getParentFragment()`